### PR TITLE
[cordova][XWALK-4055] pack cordova 4.0 using cli way

### PR DESCRIPTION
**Pull request by zhuyongyong**
- Check pack.py in usecase-cordova-android-tests, cordova-sampleapp-android-tests successfully
- Check pack_cordova_sample.py with mobilespec, helloworld, remotedebugging and gallery successfully
- Check pack.sh in webapi-noneservice-tests and webapi-service-tests successfully

BUG=https://crosswalk-project.org/jira/browse/XWALK-4055

### CODE REVIEW

File: tools/build/pack.py [hunk 1]
```diff
@@ -589,6 +588,84 @@ def packAPK(build_json=None, app_src=None, app_dest=None, app_name=None):
...
+        os.chdir(orig_dir)
+        return False
+
+    os.chdir(project_root)
+    pack_cmd = "cordova platform add android"
+    if not doCMD(pack_cmd, DEFAULT_CMD_TIMEOUT):
+        os.chdir(orig_dir)
+        return False
+
+    os.chdir(project_root)
```
Comment (Honry) [inline tools/build/pack.py]: Remove this duplicate cmd.

File: misc/webapi-noneservice-tests/pack.sh [hunk 2]
```diff
@@ -59,7 +62,11 @@ echo "        ]
 
 if [ $pack_type == "cordova" ]; then
     for suite in $LIST;do
-        python $SRC_ROOT/../../tools/build/pack.py -t ${pack_type}-aio -m $pack_mode -d $BUILD_DEST -s $SRC_ROOT/../../webapi/`basename $suite`
+        if [$sub_version == "4.0"]; then
+            python $SRC_ROOT/../../tools/build/pack.py -t ${pack_type}-aio -m $pack_mode -d $BUILD_DEST --sub-version $sub_version -s $SRC_ROOT/../../webapi/`basename $suite`
```
Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: There is no -m in Cordova 4.0 build.

Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: I guess here should be 3.6 pack command line.

Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: I think the subversion should be limited in 3.6 and 4.0.

File: misc/webapi-noneservice-tests/pack.sh [hunk 3]
```diff
@@ -104,32 +111,72 @@ if [ $pack_type == "apk" ]; then
...
-    cp -a $BUILD_DEST/opt $BUILD_ROOT/cordova/$appname/assets/www
-    mv $BUILD_DEST/testlist.json $BUILD_ROOT/cordova/$appname/assets/www
-    mv $BUILD_DEST/index_tmp.html $BUILD_ROOT/cordova/$appname/assets/www/index.html
-    mv $BUILD_DEST/icon.png $BUILD_ROOT/cordova/$appname/assets/www
+    if [$sub_version == "4.0"]; then
+        cp -ar $SRC_ROOT/../../tools/cordova_plugins $BUILD_ROOT/cordova_plugins
+        cordova create $appname org.xwalk.$appname $appname
+        sed -i "s/<widget/<widget android-activityName=\"$appname\"/g" $BUILD_ROOT/$appname/config.xml
+        sed -i "s/<\/widget>/    <allow-navigation href=\"*\" \/>\\n<\/widget>/g" $BUILD_ROOT/$appname/config.xml
+
```
Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: you miss "cordova platform add android"

File: tools/build/pack.py [hunk 4]
```diff
@@ -589,6 +588,84 @@ def packAPK(build_json=None, app_src=None, app_dest=None, app_name=None):
...
+    if not doCMD(pack_cmd, DEFAULT_CMD_TIMEOUT):
+        os.chdir(orig_dir)
+        return False
+
+    outputs_dir = os.path.join(project_root, "platforms", "android", "build", "outputs", "apk")
+    apk_files = os.listdir(outputs_dir)
+
+    if BUILD_PARAMETERS.pkgarch == "x86":
+        for i_apk in apk_files:
+            if i_apk.endswith(".apk") and "x86" in i_apk:
```
Comment (Honry) [inline tools/build/pack.py]: Please limit *debug.apk as first choice.

File: misc/webapi-noneservice-tests/pack.sh [hunk 5]
```diff
@@ -30,7 +33,7 @@ done
 # clean
 function clean_workspace(){
     echo "cleaning workspace... >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
-    rm -rf $BUILD_ROOT $BUILD_DEST
+    #rm -rf $BUILD_ROOT $BUILD_DEST
```
Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: Why note this?

File: misc/webapi-noneservice-tests/pack.sh [hunk 6]
```diff
@@ -104,32 +115,95 @@ if [ $pack_type == "apk" ]; then
...
+    if [ $sub_version == "4.0" ]; then
+        echo "-->>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>--"
+        cp -ar $SRC_ROOT/../../tools/cordova_plugins $BUILD_ROOT/cordova_plugins
+        cd $BUILD_ROOT
+        cordova create $appname org.xwalk.$appname $appname
+        sed -i "s/<widget/<widget android-activityName=\"$appname\"/g" $BUILD_ROOT/$appname/config.xml
+        sed -i "s/<\/widget>/    <allow-navigation href=\"*\" \/>\\n<\/widget>/g" $BUILD_ROOT/$appname/config.xml
+
+        cp -a $BUILD_ROOT/webrunner/*  $BUILD_ROOT/$appname/www
+        mv $BUILD_ROOT/$appname/www/index.html $BUILD_ROOT/$appname/index_real.html
```
Comment (Honry) [inline misc/webapi-noneservice-tests/pack.sh]: I guess you miss 'www' directory

File: misc/webapi-service-tests/pack.sh [hunk 7]
```diff
@@ -30,7 +33,7 @@ rm -rf $dest_dir/$name-$version-$sub_version.$pack_type.zip
 # clean
 function clean_workspace(){
     echo "cleaning workspace... >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
-    rm -rf $BUILD_ROOT
+    #rm -rf $BUILD_ROOT
```
Comment (Honry) [inline misc/webapi-service-tests/pack.sh]: Why note this?

File: misc/webapi-service-tests/pack.sh [hunk 8]
```diff
@@ -42,35 +45,72 @@ cp -a $SRC_ROOT/icon.png     $BUILD_ROOT/
...
+        sed -i "s/<\/widget>/    <allow-navigation href=\"*\" \/>\\n<\/widget>/g" $BUILD_ROOT/$appname/config.xml
+        cd $BUILD_ROOT/$appname
+        cordova platform add android
+        for plugin in `ls $BUILD_ROOT/cordova_plugins`
+        do
+            cordova plugin add $BUILD_ROOT/cordova_plugins/$plugin
+        done
+
+        cd $BUILD_ROOT/$appname/www
+        cat > index.html << EOF
```
Comment (Honry) [inline misc/webapi-service-tests/pack.sh]: Have you verify putting this change of index.html here would work?

**Comment by Honry** [comment]
Please provide the UT result.

**Comment by zhuyongyong** [comment]
The cases of all-in-one package can run successfully.

**Comment by Honry** [comment]
LGTM

**Pull request merged**